### ISSUE TITLE
A critical error occurred when importing module nodes.nodes.image.image_file_iterator: module '__main__' has no attribute '__file__'

### ISSUE THREAD

**Issue by nullmastermind** [chaiNNer-org/chaiNNer#1401]
**Information:**

- Chainner version: 0.16.0
- OS: Windows 11 Python 3.9.13

**Description**
Error while I was trying to run `yarn dev`

**Logs**
```
[0] [2022-12-15 20:46:59 +0700] [17816] [ERROR] A critical error occurred when importing module nodes.nodes.image.image_file_iterator: module '__main__' has no attribute '__file__'
[0] [2022-12-15 20:46:59 +0700] [17816] [ERROR] A critical error occurred when importing module nodes.nodes.image.load_image: module '__main__' has no attribute '__file__'
[0] [2022-12-15 20:46:59 +0700] [17816] [ERROR] A critical error occurred when importing module nodes.nodes.image.paired_image_file_iterator: module '__main__' has no attribute '__file__'
[0] [2022-12-15 20:46:59 +0700] [17816] [ERROR] A critical error occurred when importing module nodes.nodes.image.save_image: module '__main__' has no attribute '__file__'

```

**Comment by joeyballentine** [comment]
Just to confirm, this is with running the current code in main?

**Comment by joeyballentine** [comment]
Ive never seen this error before. @RunDevelopment @theflyingzamboni could this be related to the recent refactoring?

**Comment by theflyingzamboni** [comment]
I don't think so? I made sure that all of the nodes were appearing in the sidebar before the PR was merged, and one of the chains I tested used an image iterator and worked fine.

**Comment by theflyingzamboni** [comment]
@bluevariant Maybe it's an issue with using yarn? Have you tried npm instead?

**Comment by joeyballentine** [comment]
idk why yarn would cause an issue since that's on the JS side. this is a python problem. But, it is a variable so it might be worth trying with npm

**Comment by theflyingzamboni** [comment]
Same, but it was something different.

**Comment by joeyballentine** [comment]
@RunDevelopment

**Comment by RunDevelopment** [comment]
We don't use multiprocessing, so why does it not exist?

Also, I copied the `sys.modules["__main__"].__file__` trick from `add_caption`. Any fix has to be applied there as well.

**Comment by RunDevelopment** [comment]
`add_caption` uses `sys.modules["__main__"].__file__` within the function and not as a constant, so if an error occurs, you'll only see it when using the node. Could you please verify that this is the case?

---

Also, what is going on with your stack trace? From the stack trace, we can see that python starts our `run.py` in some multiprocessing manner. This is not okay since `run.py` starts a server on a specific port, and starting multiple servers listening to the same port obviously doesn't work. 

This multiprocessing doesn't happen for me. Here is what my stack trace (using an artificial error) looks like:

```
File "./run.py", line 81, in <module>
    importlib.import_module(f"{module}", package=None)
File "C:\Python38\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
File "<frozen importlib._bootstrap>", line 991, in _find_and_load
File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 783, in exec_module
File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
File "C:\Users\micha\Git\chaiNNer\backend\src\nodes\nodes\image\image_file_iterator.py", line 11, in <module>
    from ..image.load_image import ImReadNode
File "<frozen importlib._bootstrap>", line 991, in _find_and_load
File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 783, in exec_module
File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
File "C:\Users\micha\Git\chaiNNer\backend\src\nodes\nodes\image\load_image.py", line 17, in <module>
    from ...impl.dds import dds_to_png_texconv
File "<frozen importlib._bootstrap>", line 991, in _find_and_load
File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 783, in exec_module
File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
File "C:\Users\micha\Git\chaiNNer\backend\src\nodes\impl\dds.py", line 31, in <module>
    <the thing I used to cause an error>
```

So I don't understand why they are so different. Did you modify the `dev` npm script in any way?

**Comment by RunDevelopment** [comment]
> I pull `master` -> `pip install -r requirement.txt` -> `yarn install` -> `yarn dev`
> 
> Then I found the error, `Load Image` don't show
> 
> https://imgur.com/a/IcZyDAB

I just noticed this. Your screenshot shows a custom node, so it's not an unmodified version of what's in main. Are you sure that the issue is not cause by any of your modifications?

**Comment by joeyballentine** [comment]
Did you try with `npm` instead of `yarn`?

**Comment by joeyballentine** [comment]
I use windows 11 and this has never happened to me.

**Comment by green-s** [comment]
Ran into this error after upgrading to v0.17.0, using a system Python (3.10.8) with a few deps upgraded. Making a change to dds.py like the [one above](https://github.com/chaiNNer-org/chaiNNer/issues/1401#issuecomment-1356641765) fixed it, and I didn't get any error when adding a caption. However downgrading Sanic to 21.9.3 fixes it without the change. I'm guessing it's related to the [worker manager](https://sanic.dev/en/guide/deployment/manager.html) added in 22.9.

**Comment by RunDevelopment** [comment]
@joeyballentine Do we even support py 3.10 anymore? I believe I remember that there was an issue with this version.

**Comment by RunDevelopment** [comment]
The issue here was that sanic changed the default from multithreading to multiprocessing. We got the exact same error after upgrading sanic. We fixed this in #1695, so I will now close the issue.
